pym/bob/builder.py

```diff
@@ -1280,6 +1295,13 @@ async def _cookCheckoutStep(self, checkoutStep, depth):
             assert predicted, "Non-predicted incorrect Build-Id found!"
             self.__handleChangedBuildId(checkoutStep, checkoutHash)
 
+        if self.__archive.canUploadSrc(checkoutStep):
```

Comment:
So we would upload every time the checkout step is visited. This includes when the checkout was skipped! I think the upload should only be done after a fresh checkout, like the live built-Id. That may be different for bundles but otherwise it will create a lot of traffic with the artifact server.